src/Nonlinear/operators.jl

```diff
@@ -0,0 +1,748 @@
+function _create_binary_switch(ids, exprs)
+    if length(exprs) <= 3
+        out = Expr(:if, Expr(:call, :(==), :id, ids[1]), exprs[1])
+        if length(exprs) > 1
+            push!(out.args, _create_binary_switch(ids[2:end], exprs[2:end]))
+        end
+        return out
+    else
+        mid = length(exprs) >>> 1
+        return Expr(
+            :if,
+            Expr(:call, :(<=), :id, ids[mid]),
+            _create_binary_switch(ids[1:mid], exprs[1:mid]),
+            _create_binary_switch(ids[mid+1:end], exprs[mid+1:end]),
+        )
+    end
+end
+
+let exprs = map(SYMBOLIC_UNIVARIATE_EXPRESSIONS) do arg
+        return :(return $(arg[1])(x), $(arg[2]))
+    end
+    expr = _create_binary_switch(1:length(exprs), exprs)
+    @eval @inline function _eval_univariate(id, x::T) where {T}
+        $(expr)
+        return error("Invalid operator_id")
+    end
+end
+
+let exprs = map(SYMBOLIC_UNIVARIATE_EXPRESSIONS) do arg
+        if arg === :(nothing)  # f''(x) isn't defined
+            :(error("Invalid operator_id"))
+        else
+            :(return $(arg[3]))
+        end
+    end
+    expr = _create_binary_switch(1:length(exprs), exprs)
+    @eval @inline function _eval_univariate_2nd_deriv(id, x::T) where {T}
+        $(expr)
+        return error("Invalid operator_id")
+    end
+end
+
+struct _UnivariateOperator{F,F′,F′′}
+    f::F
```

Comment:
minor gripe, `\prime` looks a bit like the single quote character in the browser, and it makes the fields harder to autocomplete in the REPL

Reply:
Users should not interact with this object. There is `eval_{multi,uni}variate_{function/gradient/hessian}` instead.